src/units.cpp

```diff
@@ -464,4 +464,99 @@ double Units::scalingFactor(const UnitsPtr &units1, const UnitsPtr &units2)
+            auto unit = unitMap.find(units->name());
+            unit->second += exp;
+        }
+
+    } else if (isStandardUnitName(units->name())) {
+        auto unit = standardUnitsList.find(units->name());
+        for (const auto &u : unit->second) {
+            if (unitMap.find(u.first) == unitMap.end()) {
+                unitMap.emplace(u.first, u.second);
+            } else {
+                auto ut = unitMap.find(u.first);
+                ut->second += u.second * exp; // adding associated base exponent
+            }
+        }
+
+    } else {
+        for (size_t i = 0; i < units->unitCount(); ++i) {
+            std::string ref;
+            std::string pre;
+            std::string id;
+            double expMult;
+            units->unitAttributes(i, ref, pre, exp, expMult, id);
+            if (isStandardUnitName(ref)) {
+                auto unit = standardUnitsList.find(ref);
+                for (const auto &u : unit->second) {
+                    if (unitMap.find(u.first) == unitMap.end()) {
+                        unitMap.emplace(u.first, u.second);
+                    } else {
+                        auto ut = unitMap.find(u.first);
+                        ut->second += u.second * exp;
+                    }
+                }
+            } else {
+                auto model = owningModel(units);
+                if (model != nullptr) {
+                    auto refUnits = model->units(ref);
+                    if ((refUnits == nullptr) || refUnits->isImport()) {
+                        unitMap.clear();
+                        break;
+                    }
+                    createUnitMap(refUnits, unitMap, exp); // recursively add units to the map
+                }
+            }
+        }
+    }
+}
+
+bool Units::isEquivalentTo(const UnitsPtr &units1, const UnitsPtr &units2)
+{
+    // initial checks
+    if ((units1 == nullptr) || (units2 == nullptr)) {
+        return false;
+    }
+    if ((units1->isImport()) || (units2->isImport())) {
+        return false;
+    }
+
+    std::map<std::string, double> units1Map;
+    createUnitMap(units1, units1Map);
+    std::map<std::string, double> units2Map;
+    createUnitMap(units2, units2Map); // creating the maps to compare units over
+
+    if (units1Map.size() == units2Map.size()) {
+        auto it = units1Map.begin();
+
+        while (it != units1Map.end()) {
+            std::string unit = it->first;
+            auto found = units2Map.find(unit);
+
+            if (found == units2Map.end()) {
+                return false;
+            }
+            if ((found->second < it->second) || (found->second > it->second)) {
+                return false;
+            }
+
+            it++;
+        }
+        return true;
+    }
+    return false;
+}
+
+bool Units::isDimensionallyEquivalentTo(const UnitsPtr &units1, const UnitsPtr &units2)
+{
+    return ((Units::isEquivalentTo(units1, units2)) && (Units::scalingFactor(units1, units2) == 1.0));
```

Comment:
I feel we have gone a bit heavy on the braces here.

Reply:
Indeed, I would go with:
```
return Units::isEquivalentTo(units1, units2) && (Units::scalingFactor(units1, units2) == 1.0); 
```